python/cudf/cudf/core/column/column.py

```diff
@@ -1148,38 +1142,13 @@ def _fill(
             self.set_base_mask(mask)
 
         with acquire_spill_lock():
-            # Create a pylibcudf column to modify in-place
-            plc_col = self.to_pylibcudf(mode="write")
-            plc.filling.fill_in_place(
-                plc_col,
-                begin,
-                end,
-                fill_value,
-            )
-            # fill_in_place modifies the buffers but doesn't update metadata
-            # We need to recompute null_count if filling with null values
-            if not fill_value.is_valid(DEFAULT_STREAM):
```

Comment:
This simplification is now possible because of https://github.com/rapidsai/cudf/pull/20950.